Core/Tasks/Accelerometer.c

```diff
@@ -0,0 +1,139 @@
+//
+// Created by Taha Rashid on 2024-11-19.
+//
+
+//Adapted from https://github.com/raspberrypi/pico-examples/blob/master/i2c/lis3dh_i2c/lis3dh_i2c.c
+
+/**
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "pico/stdlib.h"
+#include "pico/binary_info.h"
+#include "hardware/i2c.h"
+
+//setting the pins to be GP20 and GP21
+//NOTE! These are the GP Pins, NOT the physical pins.
+#define PICO_DEFAULT_I2C_SDA_PIN 20
+#define PICO_DEFAULT_I2C_SCL_PIN 21
+
+/* Example code to talk to a LIS3DH Mini GPS module.
+
+   This example reads data from all 3 axes of the accelerometer and uses an auxiliary ADC to output temperature values.
+
+   Connections on Raspberry Pi Pico board, other boards may vary.
+
+   GPIO PICO_DEFAULT_I2C_SDA_PIN (On Pico this is 4 (physical pin 6)) -> SDA on LIS3DH board
+   GPIO PICO_DEFAULT_I2C_SCK_PIN (On Pico this is 5 (physical pin 7)) -> SCL on LIS3DH board
+   3.3v (physical pin 36) -> VIN on LIS3DH board
+   GND (physical pin 38)  -> GND on LIS3DH board
+*/
+
+// By default this device is on bus address 0x18
+
+const int ADDRESS = 0x18;
+const uint8_t CTRL_REG_1 = 0x20;
+const uint8_t CTRL_REG_4 = 0x23;
+const uint8_t TEMP_CFG_REG = 0xC0;
+
+#ifdef i2c_default
+
+void lis3dh_init() {
+    uint8_t buf[2];
+
+    // Turn normal mode and 1.344kHz data rate on
+    buf[0] = CTRL_REG_1;
+    buf[1] = 0x97;
+    i2c_write_blocking(i2c_default, ADDRESS, buf, 2, false);
+
+    // Turn block data update on (for temperature sensing)
+    buf[0] = CTRL_REG_4;
+    buf[1] = 0x80;
+    i2c_write_blocking(i2c_default, ADDRESS, buf, 2, false);
+
+    // Turn auxiliary ADC on
+    buf[0] = TEMP_CFG_REG;
+    buf[1] = 0xC0;
+    i2c_write_blocking(i2c_default, ADDRESS, buf, 2, false);
+}
+
+void lis3dh_calc_value(uint16_t raw_value, float *final_value, bool isAccel) {
+    // Convert with respect to the value being temperature or acceleration reading
+    float scaling;
+    float senstivity = 0.004f; // g per unit
+
+    if (isAccel == true) {
+        scaling = 64 / senstivity;
+    } else {
+        scaling = 64;
+    }
+
+    // raw_value is signed
+    *final_value = (float) ((int16_t) raw_value) / scaling;
+}
+
+void lis3dh_read_data(uint8_t reg, float *final_value, bool IsAccel) {
+    // Read two bytes of data and store in a 16 bit data structure
+    uint8_t lsb;
+    uint8_t msb;
+    uint16_t raw_accel;
+    i2c_write_blocking(i2c_default, ADDRESS, &reg, 1, true);
+    i2c_read_blocking(i2c_default, ADDRESS, &lsb, 1, false);
+
+    reg |= 0x01;
+    i2c_write_blocking(i2c_default, ADDRESS, &reg, 1, true);
+    i2c_read_blocking(i2c_default, ADDRESS, &msb, 1, false);
+
+    raw_accel = (msb << 8) | lsb;
+
+    lis3dh_calc_value(raw_accel, final_value, IsAccel);
+}
+
+#endif
+
+void InitAccelerometer() {
```

Comment:
This loop should be moved into a task file.
For example, look at the current sensor task.

The usual structure is 1 file to talk to the hardware, another file uses it.
That way, we make a seperation between the device itself and the usage

Reply:
In other words, the features that talk to the accelerometer should be in Core/UserDrivers/accelerometer.c